### PR TITLE
Client - Improve layout of filter form #99 

### PR DESCRIPTION
**Pull request by luorlandini**
The filter form should not resize and keep a fixed width and change its own position only after a defined breakpoint in addition the button to apply/clear the filters should be always visible.

the improvements also include the mobile solution

https://user-images.githubusercontent.com/4085786/116984507-9bf81f00-accb-11eb-9a87-f0d6078f4862.mov

### CODE REVIEW

File: geonode_mapstore_client/client/js/routes/Home.jsx [hunk 1]
```diff
@@ -260,6 +260,8 @@ function Home({
             ...params,
             ...newParams
         }, pathname);
+
+        (isSmallDevice) ? handleShowFilterForm() : undefined;
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/routes/Home.jsx:264]: Please remove this statement and use explicit handler for filter apply and filter clear on the filter form. Add also some comment to explain the reason.

File: geonode_mapstore_client/client/themes/geonode/scss/_filter-form.scss [hunk 2]
```diff
@@ -28,14 +27,31 @@
...
     .gn-filter-form-footer {
+        width: 20.3em;
         display: flex;
         flex-direction: row-reverse;
         align-items: center;
-        margin-bottom: 0.5rem;
-        @media screen and (max-width: 768px) {
-            margin-top: 5rem;
+        margin-bottom: 1.5rem;
+        position: fixed;
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/themes/geonode/scss/_filter-form.scss:35]: we should not use position fixed for the footer if possible, see my general comment

**Review by allyoucanmap — CHANGES_REQUESTED** [review]
There are some issue in chrome on windows 
image
image


I would like to propose a different css structure for the form:

- the from container should adapt to the container provided by the home page (no padding or margin to this node)
```css
display: flex;
flex-direction: column;
position: relative;
width: 100%;
height: 100%;
```
- form header and footer should apply a small padding 0.5rem and use small buttons size
- form body should be a flex: 1 to fill the space between the header and the footer
```css
flex: 1;
position: relative;
width: 100%;
overflow-y: auto;
padding: 0.5rem; /* not sure if it should be here or in the content node */
```
- the form content should use a position absolute to trigger the overflow-y of its own parent node
```css
position: absolute;
width: 100%;
```

scheme:

image

Please clean up the css with styles related to the form that are not used anymore such as the margin at the bottom

image

File: geonode_mapstore_client/client/js/components/home/FilterForm.jsx [hunk 3]
```diff
@@ -99,91 +99,94 @@ function FilterForm({
...
+
+                            </Col>
+                        </Form.Row>
+                    </Form>
+                </div>
             </div>
             <div className="gn-filter-form-footer">
                 <Button
-                    variant="primary"
+                    variant="primary btn-sm"
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/components/home/FilterForm.jsx:183]: you can use the prop `size="sm"`

Comment (luorlandini) [reply geonode_mapstore_client/client/js/components/home/FilterForm.jsx:183]: ok

**Review by allyoucanmap — APPROVED** [review]

**Pull request merged**